cql3/expr/prepare_expr.cc

```diff
@@ -1329,6 +1329,10 @@ binary_operator prepare_binary_operator(binary_operator binop, data_dictionary::
     auto& prepared_lhs = *prepared_lhs_opt;
     lw_shared_ptr<column_specification> lhs_receiver = get_lhs_receiver(prepared_lhs, table_schema);
 
+    if (type_of(prepared_lhs)->references_duration() && is_slice(binop.op)) {
+        throw exceptions::invalid_request_exception(fmt::format("Duration type is unordered for {}", lhs_receiver->name));
+    }
+
```

Comment:
I wonder if we also check types that contain duration, like `list<duration>`.
Maybe they fail at runtime?

Reply:
references_duration() recurses into collection types and checks the element types.